### PR TITLE
feat: Update deployment object when adapter changes are detected

### PR DESCRIPTION
**Pull request by bangqipropel**
**Reason for Change**:
1. Update deployment object when adapter changes are detected
2. Include logic for comparing adapters 
3. Add unit test for deployment update

### CODE REVIEW

File: pkg/controllers/workspace_controller.go [hunk 1]
```diff
@@ -278,6 +318,35 @@ func computeHash(w *kaitov1alpha1.Workspace) string {
 	return hex.EncodeToString(hasher.Sum(nil))
 }
 
+func compareAdapters(oldAdapters, newAdapters []kaitov1alpha1.AdapterSpec) bool {
```
Comment (Fei-Guo) [inline pkg/controllers/workspace_controller.go:321]: Add sufficient UT for this method.

File: pkg/controllers/workspace_controller.go [hunk 2]
```diff
@@ -253,6 +269,30 @@ func (c *WorkspaceReconciler) updateControllerRevision(ctx context.Context, wObj
...
+	if wObj.Inference != nil {
+		if previousWObj != nil && !compareAdapters(previousWObj.Inference.Adapters, wObj.Inference.Adapters) {
+			deployment := &appsv1.Deployment{}
+			if err := c.Get(ctx, types.NamespacedName{
+				Name:      wObj.Name,
+				Namespace: wObj.Namespace,
+			}, deployment); err != nil {
+				if !errors.IsNotFound(err) {
+					klog.ErrorS(err, "failed to get deployment", "deplyment", wObj.Name)
+				}
```
Comment (Fei-Guo) [inline pkg/controllers/workspace_controller.go:281]: You can add an annotation with revhash in the deployment, if the revhash is equal to the hash you plan to apply, you can skip the update.

**Review by Fei-Guo — COMMENTED** [review]
We need to consider how the workspace status may change when we allow to update the inference spec. This can be done in next change.

File: pkg/controllers/workspace_controller.go (hunk 2 above)
Comment (bangqipropel) [reply pkg/controllers/workspace_controller.go:281]: I can add it, but if the code is going to this position but still find the same revHash, it must have some errors, so I am reporting an error here

**Review by Fei-Guo — APPROVED** [review]

**Pull request merged**